pkg/kubelet/server.go

```diff
@@ -124,35 +123,127 @@ func NewServer(host HostInterface, enableDebuggingHandlers bool) Server {
 	return server
 }
 
-// InstallDefaultHandlers registers the default set of supported HTTP request patterns with the mux.
+// InstallDefaultHandlers registers the default set of supported HTTP request
+// patterns with the restful Container.
 func (s *Server) InstallDefaultHandlers() {
-	healthz.InstallHandler(s.mux,
+	healthz.InstallHandler(s.cont,
 		healthz.PingHealthz,
 		healthz.NamedCheck("docker", s.dockerHealthCheck),
 		healthz.NamedCheck("hostname", s.hostnameHealthCheck),
 		healthz.NamedCheck("syncloop", s.syncLoopHealthCheck),
 	)
-	s.mux.HandleFunc("/pods", s.handlePods)
-	s.mux.HandleFunc("/stats/", s.handleStats)
-	s.mux.HandleFunc("/spec/", s.handleSpec)
+	var ws *restful.WebService
+	ws = new(restful.WebService)
+	ws.
+		Path("/pods").
+		Produces(restful.MIME_JSON)
+	ws.Route(ws.GET("").
+		To(s.getPods).
+		Operation("getPods").
+		Writes([]*api.Pod{}))
+	s.cont.Add(ws)
+
+	ws = new(restful.WebService)
+	ws.
+		Path("/stats").
+		Produces(restful.MIME_JSON)
+	ws.Route(ws.GET("").
+		To(s.getStats).
+		Operation("getStats").
+		Writes([]interface{}{}))
+	ws.Route(ws.POST("").
```

Comment:
I didn't log anything, I just looked at the code that uses this endpoint.

For example, here is a GET:
https://github.com/kubernetes/kubernetes/blob/f21a6e9a93b4c928e4f598f6a748fbc99f008dd8/pkg/client/containerinfo.go#L84

And here is a POST:
https://github.com/kubernetes/heapster/blob/83c48506a53489cd8ed626b5669627ae442900be/sources/datasource/kubelet.go#L142